src/catalog/error.rs

```diff
@@ -106,6 +107,9 @@ pub enum Error {
 
     #[error("The --catalog was not built by this version of flowctl. Catalog version: '{0}'")]
     CatalogVersionMismatch(String),
+
+    #[error(transparent)]
```

Comment:
nit: sigh this was already a dumping ground, but it's getting worse... 

No action for this PR

Reply:
yeah I feel the same. I think it'll make more sense to revisit this as we break things up into separate crates. Specifically, I'm thinking about a possible factoring that separates the code for reading vs building a catalog, since most downstream crates will only need the read functionality.